google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/resources/com/google/cloud/spanner/jdbc/ClientSideStatements.json

```diff
@@ -0,0 +1,246 @@
+{
+"statements":
+	[
+	{
+	  "name": "SHOW VARIABLE AUTOCOMMIT",
+	  "executorName": "ClientSideStatementNoParamExecutor",
+	  "resultType": "RESULT_SET",
+	  "regex": "(?is)\\A\\s*show\\s+variable\\s+autocommit\\s*\\z",
+	  "method": "statementShowAutocommit",
+	  "exampleStatements": ["show variable autocommit"]
+	},
+	{
+	  "name": "SHOW VARIABLE READONLY",
+	  "executorName": "ClientSideStatementNoParamExecutor",
+	  "resultType": "RESULT_SET",
+	  "regex": "(?is)\\A\\s*show\\s+variable\\s+readonly\\s*\\z",
+	  "method": "statementShowReadOnly",
+	  "exampleStatements": ["show variable readonly"]
+	},
+	{
+	  "name": "SHOW VARIABLE RETRY_ABORTS_INTERNALLY",
+	  "executorName": "ClientSideStatementNoParamExecutor",
+	  "resultType": "RESULT_SET",
+	  "regex": "(?is)\\A\\s*show\\s+variable\\s+retry_aborts_internally\\s*\\z",
+	  "method": "statementShowRetryAbortsInternally",
+	  "exampleStatements": ["show variable retry_aborts_internally"],
+	  "prerequisiteStatements": ["set readonly=false", "set autocommit=false"]
+	},
+	{
+	  "name": "SHOW VARIABLE AUTOCOMMIT_DML_MODE",
+	  "executorName": "ClientSideStatementNoParamExecutor",
+	  "resultType": "RESULT_SET",
+	  "regex": "(?is)\\A\\s*show\\s+variable\\s+autocommit_dml_mode\\s*\\z",
+	  "method": "statementShowAutocommitDmlMode",
+	  "exampleStatements": ["show variable autocommit_dml_mode"]
+	},
+	{
+	  "name": "SHOW VARIABLE STATEMENT_TIMEOUT",
+	  "executorName": "ClientSideStatementNoParamExecutor",
+	  "resultType": "RESULT_SET",
+	  "regex": "(?is)\\A\\s*show\\s+variable\\s+statement_timeout\\s*\\z",
+	  "method": "statementShowStatementTimeout",
+	  "exampleStatements": ["show variable statement_timeout"]
+	},
+	{
+	  "name": "SHOW VARIABLE READ_TIMESTAMP",
+	  "executorName": "ClientSideStatementNoParamExecutor",
+	  "resultType": "RESULT_SET",
+	  "regex": "(?is)\\A\\s*show\\s+variable\\s+read_timestamp\\s*\\z",
+	  "method": "statementShowReadTimestamp",
+	  "exampleStatements": ["show variable read_timestamp"],
+	  "prerequisiteStatements": ["set readonly = true", "SELECT 1 AS TEST"]
+	},
+	{
+	  "name": "SHOW VARIABLE COMMIT_TIMESTAMP",
+	  "executorName": "ClientSideStatementNoParamExecutor",
+	  "resultType": "RESULT_SET",
+	  "regex": "(?is)\\A\\s*show\\s+variable\\s+commit_timestamp\\s*\\z",
+	  "method": "statementShowCommitTimestamp",
+	  "exampleStatements": ["show variable commit_timestamp"],
+	  "prerequisiteStatements": ["update foo set bar=1"]
```

Comment:
Just to confirm my understanding, these are returned on `getPrerequisiteStatements` as examples of what needs to be called beforehand? If yes, can we call these `examplePrerequisiteStatements`?

Reply:
Your assumption is correct, and these example prerequisite statements are used by some automated test cases that execute the example statements. I've changed the name of the field and the getter.